kotlinpoet/src/main/java/com/squareup/kotlinpoet/DelicateKotlinPoetApi.kt

```diff
@@ -0,0 +1,17 @@
+package com.squareup.kotlinpoet
+
+/**
+ * Marks declarations in the KotlinPoet API that are **delicate** &mdash;
+ * they have limited use-case and shall be used with care in general code.
+ * Any use of a delicate declaration has to be carefully reviewed to make sure it is
+ * properly used and does not create problems like lossy Java -> Kotlin type parsing.
+ * Carefully read documentation and [message] of any declaration marked as `DelicateKotlinPoetApi`.
+ */
+@MustBeDocumented
+@Retention(value = AnnotationRetention.BINARY)
```

Comment:
Do we need it to be binary?

Reply:
I believe so, otherwise it wouldn't be visible to consumers right?

Reply:
Merging as-is but can followup if needed!

Reply:
IIRC you only need an annotation in the bytecode if it's supposed to be queried at runtime through reflection, which isn't the case here, the annotation is only processed by the compiler

Reply:
I think you're thinking of RUNTIME retention for that case. BINARY still ensures it's in bytecode, it just can't be queried at runtime. SOURCE is like it sounds - only in source and omitted everywhere else

Reply:
True, sorry!